src/theory/quantifiers/sygus/sygus_pbe.cpp

```diff
@@ -410,19 +410,29 @@ bool CegConjecturePbe::constructCandidates(const std::vector<Node>& enums,
   Assert( enums.size()==enum_values.size() );
   if( !enums.empty() ){
     unsigned min_term_size = 0;
-    std::vector< unsigned > enum_consider;
     Trace("sygus-pbe-enum") << "Register new enumerated values : " << std::endl;
-    for( unsigned i=0; i<enums.size(); i++ ){
-      Trace("sygus-pbe-enum") << "  " << enums[i] << " -> " << enum_values[i] << std::endl;
+    std::vector<unsigned> szs;
+    for (unsigned i = 0, esize = enums.size(); i < esize; i++)
+    {
+      Trace("sygus-pbe-enum") << "  " << enums[i] << " -> ";
+      TermDbSygus::toStreamSygus("sygus-pbe-enum", enum_values[i]);
+      Trace("sygus-pbe-enum") << std::endl;
       unsigned sz = d_tds->getSygusTermSize( enum_values[i] );
+      szs.push_back(sz);
       if( i==0 || sz<min_term_size ){
-        enum_consider.clear();
         min_term_size = sz;
-        enum_consider.push_back( i );
-      }else if( sz==min_term_size ){
+      }
+    }
+    unsigned allowDiff = options::sygusPbeMultiFair() ? 0 : 1;
```

Comment:
Why is the else case like that? Which behavior is being modeled and why?

Reply:
Which else case? Do you mean what is the meaning of allowDiff?

Reply:
The case in which allowDiff is assigned to 1. I did not quite understand why "1", since it seems this only allows, when we are not "being fair", that only terms which are only one size bigger than the minimum can be added.